### PR TITLE
Adding flag to pxt serve to pick a browser

### PR DESCRIPTION
**Pull request by riknoll**
@microsoftsam would you mind adding the OSX locations for Firefox and Safari?

### CODE REVIEW

File: Jakefile.js [hunk 1]
```diff
@@ -288,6 +288,9 @@ task('serve', ['default'], { async: true }, function () {
     else if (process.env.packaged === 'true') {
         cmdArg = '-pkg'
     }
+    if (process.env.browser) {
```
Comment (pelikhan) [inline Jakefile.js]: could we prefix all env variables with "PXT_"?

File: cli/cli.ts [hunk 2]
```diff
@@ -1678,9 +1678,20 @@ export function serveAsync(...args: string[]) {
...
+                return trimmedArgs[i + 1];
+            }
+        }
+        return undefined;
+    };
+
     let justServe = false
     let packaged = false
     let includeSourceMaps = false;
+    let browser: string = argValue("browser");
```
Comment (pelikhan) [inline cli/cli.ts]: needs to be documented

File: cli/server.ts [hunk 3]
```diff
@@ -571,16 +571,56 @@ function openUrl(startUrl: string) {
...
+    else {
+        child_process.exec(`${cmds[process.platform]} ${startUrl}`);
+    }
+}
+
+
+function getBrowserLocation(browser: string) {
+    let browserPath: string;
+
+    if (browser === "chrome") {
```
Comment (pelikhan) [inline cli/server.ts]: normalize "browser" to lowercase?

File: Jakefile.js (hunk 1 above)
Comment (riknoll) [inline Jakefile.js]: Those aren't actual environement variables, it's just how jake does argument parsing. See here: http://www.jakejs.com/docs#tasks_passing_parameters_to_jake

File: cli/server.ts [hunk 4]
```diff
@@ -571,16 +571,56 @@ function openUrl(startUrl: string) {
...
+                break;
+            case "linux":
+                browserPath = "/opt/google/chrome/chrome";
+                break;
+            default:
+                break;
+        }
+    }
+    else if (browser === "firefox") {
+        browserPath = "C:/Program Files (x86)/Mozilla Firefox/firefox.exe";
```
Comment (pelikhan) [inline cli/server.ts]: firefox on other platforms?

File: cli/server.ts [hunk 5]
```diff
@@ -571,16 +571,56 @@ function openUrl(startUrl: string) {
...
+        switch (os.platform()) {
+            case "win32":
+            case "win64":
+                browserPath = "C:/Program Files (x86)/Google/Chrome/Application/chrome.exe";
+                break;
+            case "darwin":
+                browserPath = "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome";
+                break;
+            case "linux":
+                browserPath = "/opt/google/chrome/chrome";
```
Comment (pelikhan) [inline cli/server.ts]: does this work in raspbian?

File: cli/cli.ts (hunk 2 above)
Comment (riknoll) [inline cli/cli.ts]: Documented where?

**Pull request merged**